hpOneView/common.py

```diff
@@ -427,30 +431,88 @@ def make_enet_settings(name,
     }
 
 
-def make_storage_vol_templateV3(name,
-                              capacity,
-                              shareable,
-                              storagePoolUri,
-                              state='Normal',
-                              description='',
-                              storageSystemUri=None,
-                              snapshotPoolUri=None,
-                              provisionType='Thin'):
+def make_storage_volume_template_v3(name, capacity, sto_pool_uri, shareable=True, provision_type='Thin',
+                                    description='', sto_system_uri=None, snapshot_pool_uri=None):
     return {
-        'provisioning': {
-            'shareable': shareable,
-            'provisionType': provisionType,
-            'capacity': capacity,
-            'storagePoolUri': storagePoolUri},
         'name': name,
-        'state': state,
         'description': description,
-        'storageSystemUri': storageSystemUri,
-        'snapshotPoolUri': snapshotPoolUri,
+        'provisioning': {
+            'shareable': shareable,
+            'provisionType': provision_type,
+            'capacity': int(capacity)*1024*1024*1024,
+            'storagePoolUri': sto_pool_uri
+        },
+        'storageSystemUri': sto_system_uri,
+        'snapshotPoolUri': snapshot_pool_uri,
         'type': 'StorageVolumeTemplateV3'
     }
 
 
+def make_storage_volume_v3(name=None, capacity=0, sto_pool_uri=None, shareable=None, is_permanent=True,
+                           provision_type=None, description='', snapshot_uri=None, snapshot_pool_uri=None,
+                           sto_system_uri=None, sto_system_volume_name=None, template_uri=None, wwn=None):
+
+    volume = {
+        'description': description,
+        'isPermanent': is_permanent,
+        'storageSystemUri': sto_system_uri,
+        'snapshotPoolUri': snapshot_pool_uri,
+        'type': 'AddStorageVolumeV3'
+    }
+
+    if template_uri:
+        volume['name'] = name
+        volume['templateUri'] = template_uri
+
+        if capacity:
+            volume['provisioningParameters'] = {
+                'requestedCapacity': int(capacity)*1024*1024*1024
+            }
+        elif shareable and not capacity:
```

Comment:
I don't understand why you need this middle branch.  Should this be just an if and not an elif